sjsonnet/src/sjsonnet/StaticOptimizer.scala

```diff
@@ -16,6 +18,11 @@ class StaticOptimizer(ev: EvalScope, std: Val.Obj) extends ScopedExprTransform {
     } else throw e
   }
 
+  // HashMap to deduplicate strings.
+  private[this] val strings = new mutable.HashMap[String, String]
+
+  private[this] val fieldSet = new mutable.HashMap[Val.StaticObjectFieldSet, java.util.LinkedHashMap[String, java.lang.Boolean]]
+
```

Comment:
Why is this cache separate from the `Parser`'s? Should it be a single cache at the `Interpreter` level?

Reply:
Changed, there's now a single cache at the interpreter level.